internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -72,8 +73,10 @@ func (r *ResourceRender) Service() (*corev1.Service, error) {
 			if port.Protocol == ir.UDPProtocolType {
 				protocol = corev1.ProtocolUDP
 			}
+			// Listeners on merged gateways will have a port name {GatewayNamespace}/{GatewayName}/{ListenerName}.
```

Comment:
doesnt this need to be hashed to ensure length is under 15 ?

Reply:
AFAIK, Service port names apply under standard DNS label names, then it should be fine

Reply:
that means the length is 63 ? but im guessing the individual max length of `GatewayNamespace` , `GatewayName` & `ListenerName` is 63 so this need to be trimmed

Reply:
That should be trimmed as well, let's just reuse ExpectedResourceHashedName for it